environment_setup/setup_software.sh

```diff
@@ -59,6 +59,26 @@ do
     fi
 done
 
+echo "================================================================"
+echo "Installing Dependecies"
+echo "================================================================"
+
+sudo apt-get update
+
+dependencies=(
+    curl
+    cmake
+)
```

Comment:
Looks like we already have this type of list down below (called `host_software_packages`). Could we just add `curl` and `cmake` to it instead?

Reply:
I think that's installed after bazel

Reply:
Yup! I think we can just move the bazel installation to after the section where we install `host_software_packages`?

Reply:
Just trying to avoid having two different lists for installing packages from the same source (ubuntu package manager) if we don't need to.

Reply:
Yup I considered that

Reply:
I imagine you're proposing moving the `Installing Misc. Utilities` above `Installing Bazel`.  The problem is that `Installing Misc. Utilities` itself uses curl

Reply:
Ah I see. Let's just add `sudo apt-get install curl` before the bazel install then? `cmake` can still be part of the normal list of dependencies below?

Reply:
Alternatively, we actually don't need to install node or yarn now, so you can actually remove all usages of `curl` in `Installing Misc. Utilities`, then move bazel installation to after it?

Reply:
done

Reply:
Awesome thanks!